app/models/donation.rb

```diff
@@ -122,7 +122,7 @@ def replace_increase!(new_donation_params)
       original_storage_location.decrease_inventory(old_data)
     end
   rescue ActiveRecord::RecordInvalid
-    false
+    return false
```

Comment:
This shouldn't be necessary - in Ruby, the last statement in a method is automatically returned.